### DISCUSSION TITLE
Getting TPM stats from a read replica

### DISCUSSION THREAD

**Discussion by JP-Reddy** [TPC-Council/HammerDB#449] [Q&A] [answered]
Hi, 

I'm trying to test read replicas of SQL DB by running a custom read workload. So far, I've been able to run the workload and even get the time profile stats for the workload by modifying driver script and time profile code. But I also  need TPM stats from the read replica. Now, when I run the workload, I'm only able to get the TPM stats of the primary. What changes would I need to make to be able to get the stats from read replica/readable secondaries?

Thanks

**Comment by sm-shaw** [comment, marked as answer]
Hi, If you use the connect pool feature as described here https://www.hammerdb.com/blog/uncategorized/hammerdb-v4-0-new-features-pt4-connect-pooling-for-clusters/, https://hammerdb.com/docs/ch04s06.html#d0e2280

Then you will see a break down of the transactions processed by each client. Note that this is more of a guide as it shows all of the transactions including the rampup period as well. 

```
In addition to the CLIENT SIDE TPM each Virtual User will also report the total number of transactions that it processed from the time that it started running to the end of the test.

Vuser 2:VU2 processed neworder 275335 payment 273822 delivery 27495 stocklevel 27588 orderstatus 27568 transactions
Vuser 3:VU3 processed neworder 272901 payment 273475 delivery 27493 stocklevel 27194 orderstatus 27097 transactions
```
